File: 1502-construct-k-palindrome-strings/solution.py
class Solution(object):
    def canConstruct(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: bool
        """
        if k == len(s):
            return True
        elif k > len(s):
            return False
        
        charCount = [0 for _ in range(26)]
        
        for c in s:
            charCount[ord(c) - ord('a')] += 1
        
        midAligned = 0
        for c in charCount:
            if c%2 == 1:
                midAligned += 1
        
        if midAligned <= k:
            return True
        return False
        
        
        
        

File: 1502-construct-k-palindrome-strings/solution.cpp
class Solution {
public:
    bool canConstruct(string s, int k) {
        if (s.size() < k) return false;

        const int num_chars = 26;
        bool char_parity[num_chars] = {false};
        for (int i = 0; i < s.size(); ++i) {
            char_parity[s[i] - 'a'] = !char_parity[s[i] - 'a'];
        }

        int num_odd_freq = 0;
        for (int i = 0; i < num_chars; ++i) {
            num_odd_freq += char_parity[i];
        }

        return num_odd_freq <= k;
    }
};
